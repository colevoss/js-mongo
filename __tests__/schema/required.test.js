const Schema = require('../../src/Schema');

describe('Schema.<type>.required', () => {
  test('Validates if value is provided for required key', () => {
    const testData = { test: 1 };

    const validated = Schema.number.required('test', testData);

    expect(validated).toBe(1);
  });

  test('Returns error if required key is not provided', () => {
    const testData = { notTest: 1 };

    const validated = Schema.number.required('test', testData);

    expect(validated).toBeInstanceOf(TypeError);
    expect(validated).toMatchSnapshot();
  });
});
