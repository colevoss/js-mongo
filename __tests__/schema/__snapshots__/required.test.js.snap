// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema.<type>.required Returns error if required key is not provided 1`] = `[TypeError: test is required but recieved undefined]`;
